File: library/src/main/java/name/zeno/android/presenter/LifecycleListener.java
package name.zeno.android.presenter;

import android.content.Intent;

/**
 * Create Date: 16/6/25
 *
 * @author 陈治谋 (513500085@qq.com)
 */
public interface LifecycleListener
{
  void onCreate();

  void onResume();

  void onActivityResult(int requestCode, int resultCode, Intent data);

  /** fragment */
  void onViewCreated();

  /** fragment */
  void onDestroyView();

  void onDestroy();
}

File: library/src/main/java/name/zeno/android/presenter/LifecycleListener.kt
package name.zeno.android.presenter

import android.content.Intent

/**
 * @author 陈治谋 (513500085@qq.com)
 * @since  16/6/25
 */
interface LifecycleListener {
  fun onStart() {}

  fun onCreate() {}

  /** fragment  */
  fun onViewCreated() {}

  fun onResume() {}

  fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {}

  fun onPause() {}

  /** fragment  */
  fun onDestroyView() {}

  fun onDestroy() {}
}
